operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java

```diff
@@ -529,17 +530,22 @@ private boolean stillHasPodPendingRestart(DomainStatus status) {
 
```

Comment:
How can you be certain that the pod will never be null?

There are a number of steps here, and I'd be inclined to use a stream to resolve them, such as:

```
      private boolean stillHasPodPendingRestart(DomainStatus status) {
        return status.getServers().stream()
              .map(this::getServerPod)
              .map(this::getLabels)
              .anyMatch(m -> m.containsKey(LabelConstants.MII_UPDATED_RESTART_REQUIRED_LABEL));
      }

      private V1Pod getServerPod(ServerStatus serverStatus) {
        return getInfo().getServerPod(serverStatus.getServerName());
      }
```

Reply:
If the pod is null getLabels() will return an empty map, and the check for containsKey will be false.
But I like your suggestion